src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexTreeAnalyzer.cs

```diff
@@ -13,10 +13,10 @@ internal static class RegexTreeAnalyzer
         public static AnalysisResults Analyze(RegexTree regexTree)
         {
             var results = new AnalysisResults(regexTree);
-            results._complete = TryAnalyze(regexTree.Root, results, isAtomicByAncestor: true);
+            results._complete = TryAnalyze(regexTree.Root, results, isAtomicByAncestor: true, isInLoop: false);
             return results;
 
-            static bool TryAnalyze(RegexNode node, AnalysisResults results, bool isAtomicByAncestor)
+            static bool TryAnalyze(RegexNode node, AnalysisResults results, bool isAtomicByAncestor, bool isInLoop)
```

Comment:
NIT: Would it be cleaner to name this `isInAncestorLoop` or `isInLoopByAncestor`? Or perhaps leave this as `isInLoop` and change line 77 below to use `isLoop`? Just asking since for atomic you make the distinction of whether atomicity is inherited or caused by the node, so just wondering if we want to match that with this.

Reply:
Although I suppose this is different than Atomic since in here the loop node itself may not be added to the _inLoops collection...

Reply:
Yeah, it really is about whether the node is inside of some other loop.  I don't personally think adding "ancestor" to the name helps here.  The place below where isInLoop is set is just setting it there to be passed in to the child, combining whether this node is in a loop with whether this node is a loop so that the child's isInLoop reflects both.

Reply:
yeah that makes sense, I initially wrote this comment before realizing that the loop node itself wouldn't get added, so it makes sense to have the same variable name.